### PR TITLE
cherry-pick: Created mocha hooks, tslint rules updated, added prolongation of TimeoutConstants for local debug, new class ProjectAndFileTestsCode, small fixes #21901

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?

- chromedriver version updated
- tslint.json fited by analogy to downstream test automation project, tslint style fixed
- moved initial test setups from CheReporter.ts to MochaHooks.ts
- added prolongation of timeout constants if new variable TS_DEBUG_MODE set as false for local debug
- new class ProjectAndFileTestsCode. Method waitWorkspaceReadinessForCheCodeEditor() was extracted and moved to it for reuse
- workaround for issue CRW-3649
- small fixes

### Screenshot/screencast of this PR

image

### How to test this PR?

Run Empty Workspace test with VS Code editor.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@eclipse-che/che-e2e",
-  "version": "7.58.1-SNAPSHOT",
+  "version": "7.59.0-SNAPSHOT",
```
Comment (dmytro-ndp) [inline tests/e2e/package.json]: Please, keep "7.58.1-SNAPSHOT" in here.

File: tests/e2e/package.json [hunk 2]
```diff
@@ -49,19 +49,19 @@
...
-    "typed-rest-client": "^1.8.9",
+    "tslint":  "^6.1.3",
+    "typed-rest-client": "1.8.5",
     "typescript": "3.9.9",
+    "monaco-page-objects": "3.1.0",
     "vscode-extension-tester-locators": "3.1.0"
   },
   "dependencies": {
-    "@eclipse-che/api": "7.58.0",
+    "@eclipse-che/api": "latest",
```
Comment (dmytro-ndp) [inline tests/e2e/package.json]: Please, keep `7.58.0` in here.

Comment (nallikaea) [reply tests/e2e/package.json]: Done

File: tests/e2e/package.json (hunk 1 above)
Comment (nallikaea) [reply tests/e2e/package.json]: Done

File: tests/e2e/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@eclipse-che/che-e2e",
-  "version": "7.58.1-SNAPSHOT",
+  "version": "7.58.0-SNAPSHOT",
```
Comment (dmytro-ndp) [inline tests/e2e/package.json:3]: `7.58.1-SNAPSHOT` is current version

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**